File: DSA_python/DSA03024 LỰA CHỌN TỐI ƯU.py
def sort_pair(A):
    return A[1]

def submit():
    n = int(input())
    A = [0]*n
    B = [0]*n

    for i in range(n):
        A[i], B[i] = map(int, input().split())

    temp = list(zip(A, B))
    temp.sort(key=sort_pair)

    prev, count = 0, 1

    for i in range(n):
        if temp[i][0] >= temp[prev][1]:
            prev = i
            count += 1

    print(count)




test = int(input())
for i in range(test):
    submit()

File: DSA/DSA03024 LỰA CHỌN TỐI ƯU.cpp
#include <bits/stdc++.h>
using namespace std;

bool cmp(pair<int, int> a, pair<int, int> b)
{
    return a.second < b.second;
}

void submit()
{
    int n;
    cin >> n;

    vector<pair<int,int>> A(n);
    for (auto &&i : A)
    {
        cin >> i.first >> i.second;
    }
    
    sort(A.begin(), A.end(), cmp);
    
    int prev = 0, count = 1;
    
    for (int i = 1; i < n; i++)
    {
        if (A[i].first >= A[prev].second)
        {
            prev = i;
            count++;
        }
    }
    
    cout << count;

    cout << endl;
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    int test;
    cin >> test;
    while (test--)
    {
        submit();
    }
    

    return 0;
}